File: frontend/src/lib/stores/CMS.ts
import { writable, type Writable } from "svelte/store";

export type ContentType = {
    name: string,
    color: string,
    
    mcIp: string,
    discordInvite: string,
    
    discordClientId: number,
    domain: string,
}

export let CMS: Writable<ContentType | null> = writable(null)

File: src/http/cms.py
from napi.http import HTTP, Reply
from nautica import Config

@HTTP.GET()
async def branding():
    return Reply(
        name = Config("clockbot")["web.name"],
        color = Config("clockbot")["web.color"],
        
        mcIp = Config("clockbot")["web.mcIp"],
        discordInvite = Config("clockbot")["web.discordInvite"],
        
        discordClientId = str(Config("clockbot")["discordOAuth.clientId"]),
        domain = Config("clockbot")["servers.self"],
    )